### PR TITLE
Freesurfer reader MGH/MGZ reader and Triange surface (white)

### PR DESCRIPTION
**Pull request by bwagnerwork**
Still question on how to setup orientation and origin.  Especially with the dead center of volume reference of the surfaces.

### CODE REVIEW

**Comment by NicolasRannou** [comment]
Awesome

Can you fix the styling issues reported in the build please?
https://travis-ci.org/FNNDSC/ami/builds/335677286?utm_source=github_status&utm_medium=notification

Running the following should fix most of them:
```
eslint src/**/*.js --fix
```

**Comment by bwagnerwork** [comment]
Cleaned up except for some oddities around some errors I throw for the freesurfer surface types that have not been written.

/home/travis/build/FNNDSC/ami/src/loaders/loaders.freesurfer.js
  102:13   error    Unexpected lexical declaration in case block  no-case-declarations
  102:13   error    Unexpected lexical declaration in case block  no-case-declarations
  102:13   error    Unexpected lexical declaration in case block  no-case-declarations

**Comment by bwagnerwork** [comment]
Fixed the eslint error and added the quad-face reader.  How would you go about specifying color/value files that should probably be read in at the same time as the surface?  For example,  adding a  cortical parcellation to a lh.pial?

**Comment by NicolasRannou** [comment]
Nice!

Does the orientation work now by any chance?

File: src/parsers/parsers.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import Dicom from './parsers.dicom';
 import Nifti from './parsers.nifti';
 import Nrrd from './parsers.nrrd';
+import Mgh from './parsers.mgh';
```
Comment (NicolasRannou) [inline src/parsers/parsers.js]: Should be beofre Nifti (alphabetical order)

File: src/parsers/parsers.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 
 export {
     Dicom as DicomParser,
     Nifti as NiftiParser,
     Nrrd as NrrdParser,
+    Mgh as MghParser,
```
Comment (NicolasRannou) [inline src/parsers/parsers.js]: Should be before Nifti (alphabetical order)

**Comment by bwagnerwork** [comment]
Ahh, Orientation.  Nope.  MGH by itself is OK.  MGH with Freesurfer overlays not so much. Unfortunately that is my main use case.  I added a flag that can be set before reading the MGH/Freesurfer files (ParsersMgh.FREESURFER_ORIENT accessible via AMI.MghParser.FREESURFER_ORIENT) so that I can switch between the two methods.  If you get a chance to look at the sample data I uploaded and see a better way to do it I'd appreciate it.

**Comment by NicolasRannou** [comment]
THanks for your help, will do now

**Comment by NicolasRannou** [comment]
FYI I think the way to go is to display the volume at the "right" location, so it overlays nicely with other volume. The loader could also provide the transform to be applied to the FS mesh to overlay nicely somehow.

**Comment by NicolasRannou** [comment]
@bwagnerwork I believed almost all the issues are fixed now - no more tricks hopefully.

The only requirement is to apply the following transform to the FS model, in order to match the volume:

```
         let RASToLPS = new THREE.Matrix4();
          const worldCenter = r1.stackHelper.stack.worldCenter();
          RASToLPS.set( -1, 0, 0, worldCenter.x,
                        0, -1, 0, worldCenter.y,
                        0, 0, 1, worldCenter.z,
                        0, 0, 0, 1);
          object.mesh.applyMatrix(RASToLPS);
          r0.scene.add(object.mesh);
```

Sill few things to figure out but minor things hopefully. The main problem was the the coordinate system was not right handed, so that is something that we need to detect in the MGH parser.

screen shot 2018-02-09 at 4 34 14 pm

**Comment by NicolasRannou** [comment]
Awesome thx a lot!

**Comment by NicolasRannou** [comment]
(did merge with the fixes)

**Comment by NicolasRannou** [comment]
Demo should be live in a bit at: https://fnndsc.github.io/ami/#viewers_quadview

Please let me know if that looks fine to you!

**Pull request closed without merging**